src/LayerEditor/ui/data/regions_model.py

```diff
@@ -15,25 +15,26 @@ def __init__(self, le_data, regions_data):
 
         for data in regions_data:
             self.add_region_from_data(data)
-        Region.none = self.regions.get(0)
+        RegionItem.none = self.regions.get(0)
 
     def add_region_from_data(self, region_data):
-        Region( self.regions,
-                region_data.color,
-                region_data.name,
-                region_data.dim,
-                region_data.mesh_step,
-                region_data.not_used,
-                region_data.boundary)
+        RegionItem(self.regions,
+                   region_data.color,
+                   region_data.name,
+                   region_data.dim,
+                   region_data.mesh_step,
+                   region_data.not_used,
+                   region_data.boundary,
+                   region_data.brep_shape_ids)
 
     @undo.undoable
     def add_region(self, name="", dim=0, color=None):
-        reg = Region(self.regions, color=color, name=name, dim=dim)
+        reg = RegionItem(self.regions, color=color, name=name, dim=dim)
```

Comment:
Rather create out of this method, less parameters to pass.

Reply:
This was also changed. Since it is undoable method it should not create the object, but should get passed finished object as parameter.